ntripclient.c

```diff
@@ -566,6 +583,104 @@ static int encode(char *buf, int size, const char *user, const char *pwd)
   return bytes;
 }
 
+static ssize_t curl_send_(CURL *curl, const void *buf, size_t len, int flags)
+{
+    (void)flags; // not used
+    size_t bytes_sent = 0;
+    CURLcode result = curl_easy_send(curl, buf, len, &bytes_sent);
+    return (result == CURLE_OK) ? (ssize_t)bytes_sent : -1;
+}
+
+static ssize_t curl_recv_(CURL *curl, void *buf, size_t len, int flags)
+{
+    (void)flags; // not used
+    size_t bytes_recv = 0;
+    CURLcode result = CURLE_OK;
+
+    do
+    {
+        result = curl_easy_recv(curl, buf, len, &bytes_recv);
+
+        // Wait for 100ms before trying again.
+        if (result == CURLE_AGAIN)
+            SLEEP_MS(100);
+    } while (result == CURLE_AGAIN);
+
+    return (result == CURLE_OK) ? (ssize_t)bytes_recv : -1;
+}
+
+static int parse_http_header(CURL *curl, HttpHeader *hdr)
+{
+    char buf[MAXDATASIZE];
+    int total_bytes = 0;
+    bool hdr_done = false;
+
+    memset((void*)hdr, 0, sizeof(HttpHeader));
+
+    while (hdr_done == false)
+    {
+        // Read more data into the buffer; append to existing data if any.
+        ssize_t bytes_read = curl_recv_(curl, buf + total_bytes, MAXDATASIZE - total_bytes - 1, 0);
+        if (bytes_read <= 0)
+            return -1;
```

Comment:
Might I suggest we use an enum for return codes like these to improve readability?
e.g. typedef enum {
    INVALID_NUM_BYTES_READ = -1,
    VALID_NUM_BYTES_READ = 0,
} ReadBufferRetCode;

and then in main(): 
`ReadBufferRetCode retcode = parse_http_header(curl, &http_hdr);`
`...`
`if (retcode != VALID_NUM_BYTES_READ || ...)`

Reply:
How about I make the return a boolean since it's a all or nothing we really care about?

Reply:
Sure that works too 👍

Reply:
Done